update KB to use coverimages

mattdias opened the pull request:
update the cover images for announcements in KB
updated the style to use BS5 cards
removed inconsistent fonts and styles

blaisefonseca commented on app/components/kb/articles-by-tag.vue:
```diff
@@ -10,23 +10,25 @@
 					<hr>
 
                     <div class="kb-listing w-100">
-                        <ul class="article-with-tags-ul">
```
can you delete the css classes as well

mattdias replied:
i think it is still being used. In another PR i would get rid the whole KB style and use BS

mattdias replied:
removed kb style from the KB.

blaisefonseca commented on app/components/kb/filters.js:
```diff
@@ -3,7 +3,7 @@ import Vue from 'Vue';
 export function formatDate(datetime, format) {
     if (format === undefined)
-        format = 'DD MMM YYYY HH:mm';
+        format = 'DD MMM YYYY';
```
pass the required format from you usage instead of changing the global filter

mattdias replied:
we already have a function formatDateWithTime what we use in other places for time.

blaisefonseca replied:
`formatDateWithTime` is in angularjs portion, this one is in vueJs

mattdias replied:
ok

blaisefonseca commented on app/css/vue/kb/style.css:
```diff
@@ -115,26 +115,26 @@
 	margin-top: 1.5rem;
 }
 .right-side-articles .view-more a{
-	color: #00405C;
+	
 }
```
remove empty class

mattdias replied:
done

blaisefonseca commented on app/css/vue/kb/style.css:
```diff
@@ -115,26 +115,26 @@
 
 .knowledge-base .categories-list .view-more:hover{
-	color: #00405C;
+	
```
remove empty lines

mattdias replied:
done

blaisefonseca commented on app/views/kb/home.vue:
```diff
@@ -1,22 +1,12 @@
                 <div class="row">
-                    <div class="col-lg-12">
-                        <header>
-                            <div class="categerios-parahraph">
```
remove css class too

mattdias replied:
this was never a class

blaisefonseca commented on app/components/kb/article-details.vue:
```diff
@@ -4,6 +4,11 @@
                 <div class="article-description mt-0 single">
+                  
+                   <div v-if="article.coverImage" class="max-height:450px;" > 
+                         <img v-bind:src="getSizedImage(article.coverImage.url, '800x800')" class="img-fluid">
```
it will be good to provide default image when cover image is not available

mattdias replied:
then we will have to have an image even if we dont want one. what should the default image be?

blaisefonseca replied:
generally simple div with light color or any blank image.

mattdias replied:
I dont think we need one.

blaisefonseca reviewed (changes requested):
see somments

mattdias commented on app/components/kb/popular-tags.vue:
```diff
@@ -4,9 +4,15 @@
         <h4>{{ $t("popularTags") }}</h4>
 				<hr>
         <div class="tag-scroll" v-if="!loading">
+            <div class="tagcloud" >
+                <a href="#" class="btn btn-mini" @click="goToAdminTag($realm.is('BCH') ? 'bch-announcement' : 'absch-announcement')">Announcements</a>
+            </div>
             <div class="tagcloud" v-for="tag in popularTags">
                 <a href="#" class="btn btn-mini" @click="goToAdminTag(tag.adminTags[0])">{{tag.title}}</a>
             </div>
+            <div class="tagcloud" >
+                <a href="#" class="btn btn-mini" @click="goToAdminTag('faq')">FAQs</a>
+            </div>
```
Then they would show up in the KB homepage

blaisefonseca replied:
push them from Javascript, what difference will it make if they show up on KB homepage?

mattdias replied:
it doesn't follow the same format on the homepage. announcements change so would need to get latest announcements dynamically.

mattdias commented on app/css/vue/kb/style.css:
```diff
@@ -405,7 +395,7 @@
 	background-color: #fff;
 	border-radius: 0.25rem;
-	padding: 1.87rem 1.87rem 0.6rem 1.87rem;
+	padding: 10px;
```
any case we should eventually remove this whole style sheet

mattdias commented:
added random shuffle to the relevant articles so not always the same. 
fixed the back button, since we don't always come from KB home page
remove links to the custom KB style sheet, replaced with bootstrap.
added dates.

blaisefonseca commented on app/components/kb/relevant-articles.vue:
```diff
@@ -64,6 +77,17 @@ export default {
             return this.getUrl(this.$options.filters.lstring(article.title), article._id, this.tag);
         },
+        shuffleArray(array) {
+            let curId = array.length;
+            while (0 !== curId) {
+                let randId = Math.floor(Math.random() * curId);
+                curId -= 1;
+                let tmp = array[curId];
+                array[curId] = array[randId];
+                array[randId] = tmp;
+            }
+            return array;
```
`return array.map((item, i)=>{ return { ...item, randomSort : Math.floor(Math.random() * 100)}}).sort((a,b)=> a.randomSort-b. randomSort)` is simple to understand

eg `[{name:'blaise'}, {name:'fonse'}, {name:'name'}].map((item, i)=>{ return { ...item, randomSort : Math.floor(Math.random() * 100)}}).sort((a,b)=> a.randomSort-b. randomSort)`

mattdias replied:
ok

blaisefonseca commented on app/components/kb/article-details.vue:
```diff
@@ -1,109 +1,115 @@
+    },
+    methods: {
+        back() {
+            window.history.back();
```
if they come from external site like google, they will go back to google.

mattdias replied:
but that is better than going back the wrong page.

blaisefonseca commented on app/components/kb/relevant-articles.vue:
```diff
@@ -1,13 +1,14 @@
-        <h4>
-            <span>{{ $t("relevantArticles") }}</span>
+    <div>
+        <h4 v-if="!hideTitle">
```
why hide the title?

if you need to override some section use slots https://vuejs.org/guide/components/slots.html#slot-content-and-outlet

mattdias replied:
im not sure what you mean. 
relevantarticles is used in side bar and has a title, but now I am  also using it on categories page when displaying latest announcements to look the same layout as the other categories data from the json

blaisefonseca reviewed (commented):
two more comments, can you in future not add new feature code to existing PR which is under review, difficult to track
